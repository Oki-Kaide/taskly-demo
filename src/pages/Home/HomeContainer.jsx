import React from "react";
import Home from "./Home";
import ProtonSDK from "../../utils/proton";

class HomeContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      windowWidth: 0,
    };
  }

  componentDidMount() {
    this.updateWindowWidth();
    window.addEventListener("resize", this.updateWindowWidth);
    /* istanbul ignore next */
    document.addEventListener("backToSelector", () => {
      this.generateLoginRequest();
    });
  }

  updateWindowWidth = () => {
    this.setState({ windowWidth: window.innerWidth });
  };

  /* istanbul ignore next */
  generateLoginRequest = async () => {
    const { setLoggedInState, setErrorState } = this.props;
    const { auth, accountData, error } = await ProtonSDK.login();
    if (error) {
      setErrorState(error);
      return;
    }
    if (!auth || !auth.actor || !auth.permission) {
      return;
    }
      
    setLoggedInState(auth.actor, auth.permission, accountData);
  };

  render() {
    const { windowWidth } = this.state;
    const { error } = this.props;

    return (
      <Home
        openLoginModal={this.generateLoginRequest}
        windowWidth={windowWidth}
        error={error}
      />
    );
  }
}

export default HomeContainer;
