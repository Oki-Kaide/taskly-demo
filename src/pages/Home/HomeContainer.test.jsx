import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomeContainer from './HomeContainer';

describe('HomeContainer component', () => {
  it('renders without crashing', () => {
    shallow(<HomeContainer />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<HomeContainer />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
