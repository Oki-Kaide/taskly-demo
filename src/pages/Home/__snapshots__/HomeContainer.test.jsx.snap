// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer component matches the rendered snapshot 1`] = `
<Home
  openLoginModal={[Function]}
  windowWidth={1024}
/>
`;
